anndata/core/anndata.py

```diff
@@ -1260,9 +1260,10 @@ def _remove_unused_categories(self, df_full, df_sub, uns):
                     # this is a strange hack...
                     if np.array(uns[k + '_colors']).ndim == 0:
                         uns[k + '_colors'] = np.array(uns[k + '_colors'])[None]
-                    uns[k + '_colors'] = np.array(uns[k + '_colors'])[
-                        np.where(np.in1d(
-                            all_categories, df_sub[k].cat.categories))[0]]
+                    elif np.array(uns[k + '_colors']).ndim > 1:
```

Comment:
why not `else`? `ndim` can hardly be `<0`, can it?

Reply:
Sure?  I no longer remember why I didn't use an `else` here.

Reply:
How would an array with -1 dimensions look like?

Reply:
https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.ndim.html